### PR TITLE
Create CI issue only when failing to build / test

### PR DESCRIPTION
**Pull request by fmauch**
Also, I've added different templates for failing the target ws vs. the downstream ws.

This should implement #28.

I haven't tested this, yet, hence the draft status. Will need another timeslot for testing this.

### CODE REVIEW

File: .github/workflows/reusable-industrial-ci-with-cache.yml [hunk 1]
```diff
@@ -105,15 +105,25 @@ jobs:
         run:
           wget https://raw.githubusercontent.com/ros-controls/ros2_control_ci/master/.github/issue_template_failed_ci.md -O .github/issue_template_failed_ci.md
       - uses: JasonEtco/create-an-issue@v2
-        if: ${{ always() && steps.ici.outcome == 'failure' && github.event_name == 'schedule' }}
+        if: ${{ always() && (steps.ici.outputs.build_target_workspace == "1" || steps.ici.outputs.run_target_test || steps.ici.outputs.target_test_results == "1" ) == 'failure' && github.event_name == 'schedule' }}
```
Comment (christophfroehlich) [inline .github/workflows/reusable-industrial-ci-with-cache.yml:108]: could you please explain this expression? it compares if the result of the 3x OR inside the parenthesis equals `'failure'`?

**Review by christophfroehlich — COMMENTED** [review]
This should not break the workflow itself, we can just try it out..

File: .github/workflows/reusable-industrial-ci-with-cache.yml (hunk 1 above)
Comment (fmauch) [reply .github/workflows/reusable-industrial-ci-with-cache.yml:108]: whoops, the `== 'failure'` probably needs to go. I'll create a branch testing this and then finish this PR today.

**Comment by fmauch** [comment]
Tested things with

- An error [during dependency installation](https://github.com/fmauch/control_toolbox/actions/runs/8139177972/job/22249998778) resulting in no issue
- An error [during build](https://github.com/fmauch/control_toolbox/actions/runs/8139183192/job/22250002694) resulting in the same issue as before
- An error [during downstream build](https://github.com/fmauch/control_toolbox/actions/runs/8141890694/job/22250206971) resulting in an [issue with another template for downstream failures](https://github.com/fmauch/control_toolbox/issues/20).

I did not test failing tests, but looking at `colcon test-result`'s [code](https://github.com/colcon/colcon-test-result/blob/63cd568fb255adedaf82ef128fb81c00928b3296/colcon_test_result/verb/test_result.py#L109) and [ici itself](https://github.com/ros-industrial/industrial_ci/blob/c2b7b619ba802b3c3ae95d400076f6e17a2d86c3/.github/workflows/main.yml#L140) that should be fine. Reverting all the temporary steps to make this mergable now.

**Review by christophfroehlich — APPROVED** [review]
thx. looks fine, let's test it live!

**Pull request merged**